### PR TITLE
Update build and release instructions

### LINKED ISSUE #71: Instructions for building
Are the [instructions for building](https://github.com/kelloggm/object-construction-checker/blob/master/README-developers.md) still up to date in light of [moving the Returns Receiver Checker into this repository](https://github.com/kelloggm/object-construction-checker/commit/d544cc3891ba9dd643e844efeb73ebfbdd83f88e)?

### PR DESCRIPTION
**Pull request by msridhar**
Fixes #71

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -1,12 +1,15 @@
 Releasing
 =========
 
+Note that these steps require pushing directly to the master branch on GitHub.  Be sure that other developers are not pushing changes concurrently.
+
  1. Change the version in `gradle.properties` to a non-SNAPSHOT version.
- 2. Update all README files with the new version.
+ 2. Update all README files (except `README-developers.md`) with the new release version.
```
Comment (mernst) [inline RELEASING.md]: I suggest combining points 1 and 2 since they are both setting a new version in some files.

File: RELEASING.md [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
  3. `git commit -am "Prepare for release X.Y.Z."` (where X.Y.Z is the new version)
```
Comment (mernst) [inline RELEASING.md]: I like the commands to be copy-and-pasteable to the extent possible.  So I would suggest adding a step

export NEWVER=X.Y.Z (where X.Y.Z is the new version)

and then the subsequent steps can use that environment variable.

File: RELEASING.md [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
  4. `git tag -a vX.Y.Z -m "Version X.Y.Z"` (where X.Y.Z is the new version)
  5. `./gradlew -Dorg.gradle.parallel=false clean uploadArchives`
  6. Update the `gradle.properties` to the next SNAPSHOT version.
- 7. `git commit -am "Prepare next development version."`
- 8. `git push && git push --tags`
- 9. Visit [Sonatype Nexus](https://oss.sonatype.org/) and promote the artifact.
+ 7. Update `README-developers.md` to refer to the new SNAPSHOT version.
+ 8. `git commit -am "Prepare next development version."`
+ 9. `git push && git push --tags`
+ 10. Visit [Sonatype Nexus](https://oss.sonatype.org/) and promote the artifact.
```
Comment (mernst) [inline RELEASING.md]: Does this require any particular credentials?

Comment (msridhar) [reply RELEASING.md]: Yes, at this point, only I can do this (we can address in the future).  These instructions are only intended for developers on the team with such credentials.

**Comment by msridhar** [comment]
Thanks, @mernst, I am going to merge for now and open a follow-up task for addressing the other feedback

**Pull request merged**